cmd/ct_monitor/main.go

```diff
@@ -0,0 +1,178 @@
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	ctgo "github.com/google/certificate-transparency-go"
+	ctclient "github.com/google/certificate-transparency-go/client"
+	"github.com/google/certificate-transparency-go/jsonclient"
+	"github.com/sigstore/rekor-monitor/pkg/ct"
+	"github.com/sigstore/rekor-monitor/pkg/identity"
+	"github.com/sigstore/rekor-monitor/pkg/notifications"
+	"github.com/sigstore/rekor-monitor/pkg/util/file"
+	"gopkg.in/yaml.v2"
+)
+
+// Default values for monitoring job parameters
+const (
+	publicRekorServerURL     = "https://rekor.sigstore.dev"
+	logInfoFileName          = "logInfo.txt"
+	outputIdentitiesFileName = "identities.txt"
+)
+
+// This main function performs a periodic identity search.
+// Upon starting, any existing latest snapshot data is loaded and the function runs
+// indefinitely to perform identity search for every time interval that was specified.
+func main() {
+	configFilePath := flag.String("config-file", "", "path to yaml configuration file containing identity monitor settings")
+	configYamlInput := flag.String("config", "", "path to yaml configuration file containing identity monitor settings")
+	once := flag.Bool("once", true, "whether to run the monitor on a repeated interval or once")
+	serverURL := flag.String("url", publicRekorServerURL, "URL to the rekor server that is to be monitored")
+	interval := flag.Duration("interval", 5*time.Minute, "Length of interval between each periodical consistency check")
+	flag.Parse()
+
+	if *configFilePath == "" && *configYamlInput == "" {
+		log.Fatalf("empty configuration input")
+	}
+
+	if *configFilePath != "" && *configYamlInput != "" {
+		log.Fatalf("only input one of configuration file path or yaml input")
+	}
+
+	var config notifications.IdentityMonitorConfiguration
+
+	if *configFilePath != "" {
+		readConfig, err := os.ReadFile(*configFilePath)
+		if err != nil {
+			log.Fatalf("error reading from identity monitor configuration file: %v", err)
+		}
+
+		configString := string(readConfig)
+		if err := yaml.Unmarshal([]byte(configString), &config); err != nil {
+			log.Fatalf("error parsing identities: %v", err)
+		}
+	}
+
+	if *configYamlInput != "" {
+		if err := yaml.Unmarshal([]byte(*configYamlInput), &config); err != nil {
+			log.Fatalf("error parsing identities: %v", err)
+		}
+	}
+
+	var fulcioClient *ctclient.LogClient
+	fulcioClient, err := ctclient.New(*serverURL, http.DefaultClient, jsonclient.Options{})
+	if err != nil {
+		log.Fatalf("getting Fulcio client: %v", err)
+	}
+
+	allOIDMatchers, err := config.MonitoredValues.OIDMatchers.RenderOIDMatchers()
+	if err != nil {
+		fmt.Printf("error parsing OID matchers: %v", err)
+	}
+
+	monitoredValues := identity.MonitoredValues{
+		CertificateIdentities: config.MonitoredValues.CertificateIdentities,
+		Subjects:              config.MonitoredValues.Subjects,
```

Comment:
We don't need subjects, correct? Since subjects are just for special cases of SSH certificates or emails in keys

Reply:
removing subjects and fingerprints from this construction